athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/handlers/GlueMetadataHandlerTest.java

```diff
@@ -100,6 +103,28 @@
 
     private BlockAllocatorImpl allocator;
 
+    // The following list is purposely unordered to demonstrate that the doListTables pagination logic does not
+    // consider the order of the tables deterministic (i.e. pagination will work irrespective of the order that the
+    // tables are returned from the source).
+    private final List<Table> unPaginatedTables = new ImmutableList.Builder<Table>()
+            .add(new Table().withName("table3"))
+            .add(new Table().withName("table2"))
+            .add(new Table().withName("table5"))
+            .add(new Table().withName("table4"))
+            .add(new Table().withName("table1"))
+            .build();
+
+    // The following response is expected be returned from doListTables when the pagination pageSize is greater than
+    // the number of tables in the unPaginatedTables list (or pageSize has UNLIMITED_PAGE_SIZE_VALUE).
+    private final ListTablesResponse fullListResponse = new ListTablesResponse(catalog,
+            new ImmutableList.Builder<TableName>()
+                    .add(new TableName(schema, "table1"))
+                    .add(new TableName(schema, "table2"))
+                    .add(new TableName(schema, "table3"))
+                    .add(new TableName(schema, "table4"))
+                    .add(new TableName(schema, "table5"))
+                    .build(), null);
+
```

Comment:
this is exactly what I asked for. I asked for us to split up the tests. Moving this that are only needed by 1 function onto the class itself does shorten that function but it doesn't accomplish the goal of having smaller independent tests.

Reply:
Broke up the tests into three: doListTablesWithUnlimitedPageSize, doListTablesWithLargePageSize, and doListTablesWithPagination.